### ISSUE TITLE
SSL_CTX_new gets "error retrieving entropy" 

### ISSUE THREAD

**Issue by anshah1216** [openssl/openssl#19824]
Using openssl-1.1.1g on Ubuntu 20.04 

Seeing getting an error when trying to create an SSL context.
```
SSL_CTX  *m_ctx;
m_ctx = SSL_CTX_new(TLS_client_method());
if (!m_ctx) 
{
    print_error("SSL_CTX_new");
    return;
}
```

In the logs this error is seen:
```
12/02/2022 22:30:52:019 Info ( 155): SSL_CTX_new() failed: error retrieving entropy
```

Here's what I found on another forum:

> Linux kernels 4.8 and later, don't have a reliable way to detect
> that /dev/urandom has been properly seeded, so a failure is raised
> for this case (i.e. the getentropy(2) call has already failed).

**Comment by mattcaswell** [comment]
Something fundamentally bad happened generating random data. Is this using the stock OpenSSL libs from Ubuntu 20.04? Or did you compile this yourself? Most commonly we see problems with random data on constrained devices...what kind of machine are your running this on?

**Comment by anshah1216** [comment]
> Something fundamentally bad happened generating random data. Is this using the stock OpenSSL libs from Ubuntu 20.04? Or did you compile this yourself? Most commonly we see problems with random data on constrained devices...what kind of machine are your running this on?

Thank you @mattcaswell  for responding.  
I compiled this myself and using the OpenSSL static libraries.
I'm running on an Ubuntu 20.04 VM with kernel: 5.4.0-77-generic

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
